fastapi_utils/tasks.py

```diff
@@ -74,7 +74,7 @@ async def loop() -> None:
                     repetitions += 1
                     await asyncio.sleep(seconds)
 
-            await loop()
+            await asyncio.ensure_future(loop())
```

Comment:
@Keiishu : I think you need to remove the `await`. This is what was used in `v0.2.1`. I test your code but remove the `await` in my local FastAPI project, and it worked. It no longer block the app to start.

Reply:
That's what I did in the first commit on this PR, and it works for me, I've used it like this with no issues whatsoever in a relatively big FastAPI project. But then, tests were not passing anymore and I thought it was my fault. So I must say I'm a little bit confused.
Maybe the tests are faulty?

Reply:
I'm not familiar with sarlette but I'm pretty confident that asyncio.ensure_future doesn't return coroutine, therefore await isn't needed. I'm using the below code since the 0.6.0 repeat_every break, and this works fine for me.

```py
def repeat(seconds: float, delay: float = None):
    def decorator(func):
        is_coroutine = asyncio.iscoroutinefunction(func)
        @wraps(func)
        async def wrapper(*args, **kwargs):
            async def repeat_function():
                if delay:
                    await asyncio.sleep(delay)

                while True:
                    if is_coroutine:
                        await func(*args, **kwargs)
                    else:
                        run_in_threadpool(func, *args, **kwargs)
                    await asyncio.sleep(seconds)

            asyncio.ensure_future(repeat_function())

        return wrapper
    return decorator
```

Reply:
@mgkid3310 You're right, `await` shouldn't be necessary. But without it, the tests introduced with 0.6.0 fail. And while your code works perfectly fine for me as well, as do the first commit of this PR, the tests all fail.
So yeah, maybe there is an issue with the tests themselves...